File: javascript/selectionSort.js
/**
 * @param {Array} Array
 * @return {Array}
 */

export default function selectionSort(ar) {
    for (let i =0; i < ar.length; i++) {
        let min = i
        for (let j = i+ 1; j < ar.length; j++) {
            if (ar[j] < ar[min]) {
                min = j
            }
        }
        let temp = ar[i]
        ar[i] = ar[min]
        ar[min] = temp
    }
    return ar
}

File: C++/SelectionSort.cpp
#include <iostream>

using namespace std;


void selectionSort(int[], int);
void swap(int &, int &);

int main()
{
    const int SIZE = 6;

    int values[SIZE] = {8,3,5,7,1,2};

    cout << "The unsorted values are:" << endl;
    for(auto element : values)
        cout << element << " ";
    cout << endl;

    selectionSort(values, SIZE);

    cout << "The sorted values: " << endl;
    for(auto element : values)
        cout << element << " ";
    cout << endl;

    return 0;
}

void selectionSort(int array[], int size)
{
    int minIndex,
            minValue;

    for(int start = 0; start < (size - 1); start++)
    {
        minIndex = start;
        minValue = array[start];
        for(int index = start + 1; index < size; index++)
        {
            if(array[index] < minValue)
            {
                minValue = array[index];
                minIndex = index;
            }
        }
        swap(array[minIndex], array[start]);
    }
}

void swap(int &a, int &b)
{
    int temp = a;
    a = b;
    b = temp;
}


